### PR TITLE
Added scope and extra keyword arguments that can be optionally passed to client crerdentials flow

### PR DESCRIPTION
**Pull request by infohash**
To comply with 4.4.2.  Access Token Request of Client Credentials Grant in  RFC 6749.

### CODE REVIEW

**Review by zamzterz — APPROVED** [review]
Great - thanks! 👍

File: src/flask_pyoidc/pyoidc_facade.py [hunk 1]
```diff
@@ -261,6 +261,13 @@ def client_credentials_grant(self):
         On API call, token introspection will ensure that only valid token can
         be used to access your APIs.
 
+        Parameters
+        ----------
+        scope: list, optional
```
Comment (zamzterz) [inline src/flask_pyoidc/pyoidc_facade.py:266]: Type should be `Optional[list]`.

File: src/flask_pyoidc/pyoidc_facade.py [hunk 2]
```diff
@@ -261,6 +261,13 @@ def client_credentials_grant(self):
...
+            List of scopes to be requested.
+        **kwargs : dict, optional
```
Comment (zamzterz) [inline src/flask_pyoidc/pyoidc_facade.py:268]: Type should be `Optional[dict]`.

File: src/flask_pyoidc/pyoidc_facade.py (hunk 1 above)
Comment (infohash) [reply src/flask_pyoidc/pyoidc_facade.py:266]: Numpy docstring format rules say to annotate docstring parameters like this only.

Comment (zamzterz) [reply src/flask_pyoidc/pyoidc_facade.py:266]: Right, I've just reviewed those and the Google Python style guide and seems it's changed compared to the current format in this code base.
I'm thinking about dropping the types in docs and moving fully to proper type annotations.

**Pull request merged**